app/src/pages/eligibility.tsx

```diff
@@ -1,76 +1,122 @@
-import { useAppContext } from '@context/state'
+import type { EligibilityData, ModifySessionProps } from '@customTypes/common'
 import type { GetServerSideProps, NextPage } from 'next'
 import { Trans } from 'next-i18next'
 import { serverSideTranslations } from 'next-i18next/serverSideTranslations'
-import { ChangeEvent, useEffect, useState } from 'react'
+import { useRouter } from 'next/router'
+import { ChangeEvent, useState } from 'react'
 
 import BackLink from '@components/BackLink'
 import ButtonLink from '@components/ButtonLink'
 import InputChoiceGroup from '@components/InputChoiceGroup'
 import RequiredQuestionStatement from '@components/RequiredQuestionStatement'
 
-interface Props {
-  previousRoute: string
-}
+const Eligibility: NextPage<ModifySessionProps> = (
+  props: ModifySessionProps
+) => {
+  // Get the session from props.
+  const { session, setSession } = props
+  // Initialize form as a state with the value in session.
+  const [form, setForm] = useState<EligibilityData>(session.eligibility)
 
-const Eligibility: NextPage<Props> = (props: Props) => {
-  const incomeRoute = '/income'
-  const { session, setSession } = useAppContext()
-  const [continueBtn, setContinueBtn] = useState({
-    labelKey: 'continue',
-    route: incomeRoute,
-  })
-  const [form, setForm] = useState(session?.eligibility)
-  const requiredMet = () => {
-    const categorical = [
-      'pregnant',
-      'baby',
-      'child',
-      'guardian',
-      'pregnant',
-      'none',
-    ].some((category) => form[category as keyof typeof form])
-    const programs = ['insurance', 'snap', 'tanf', 'none2'].some(
-      (program) => form[program as keyof typeof form]
+  // Function to check whether all the required fields in this form
+  // page have been filled out.
+  // @TODO: This could be further refactored to be more generic.
+  const isRequiredMet = (formToCheck: EligibilityData) => {
+    return (
+      formToCheck.residential !== '' &&
+      formToCheck.categorical.length > 0 &&
+      formToCheck.previouslyEnrolled !== '' &&
+      formToCheck.adjunctive.length > 0
     )
+  }
 
-    return form.residential && categorical && form.before && programs
+  // Function to update button route.
+  // Route the user to /other-benefits if they do not meet basic eligibility requirements.
+  // Route the user to /income if they do not meet adjunctive eligibility requirements.
+  // Otherwise, route the user to /choose-clinic.
+  const getRouting = (formToCheck: EligibilityData) => {
+    const isResident = formToCheck.residential === 'yes'
+    const hasCategory = !formToCheck.categorical.includes('none')
+    if (!isResident || !hasCategory) {
+      return '/other-benefits'
+    } else if (formToCheck.adjunctive.includes('none')) {
+      return '/income'
+    } else {
+      return '/choose-clinic'
+    }
```

Comment:
This will also probably get pulled out into a data validation and/or routing utils file.